lib/src/ctx/context.rs

```diff
@@ -65,16 +68,37 @@ impl<'a> Debug for Context<'a> {
 }
 
 impl<'a> Context<'a> {
+	pub(crate) fn from_ds(
+		time_out: Option<Duration>,
+		capabilities: Capabilities,
+		index_stores: IndexStores,
+	) -> Context<'a> {
+		let mut ctx = Self {
+			values: HashMap::default(),
+			parent: None,
+			deadline: None,
+			cancelled: Arc::new(AtomicBool::new(false)),
+			notifications: None,
+			query_planner: None,
+			capabilities: Arc::new(capabilities),
+			index_stores,
+		};
+		if let Some(timeout) = time_out {
+			ctx.add_timeout(timeout);
+		}
```

Comment:
Previously the Context was initialized with default capabilities and then they were immediately replaced. This constructor creates the context with the non optional initial capabilities and indexstore.